### PR TITLE
[docs] UDP commands - added Doxygen tags

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

File: src/cli/cli_udp.cpp [hunk 1]
```diff
@@ -120,6 +188,61 @@ template <> otError UdpExample::Process<Cmd("open")>(Arg aArgs[])
...
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -s 800
+ * Done
+ * @endcode
+ * @cparam udp send [@ca{ip} @ca{port}] [@ca{type}] @ca{value}
+ * Parameter definitions:
+ * - `ip`: Destination address. This address can be either an IPv4 or IPv6 address,
+ *   An IPv4 address gets synthesized to an IPv6 address with the preferred
+ *   NAT64 prefix from the network data. (If the preferred NAT64 prefix
+ *   is unavailable, the command returns `InvalidState`).
+ * - `port`: UDP destination port
```
Comment (abtink) [inline src/cli/cli_udp.cpp:236]:
```suggestion
 * - `port`: UDP destination port.
```

File: src/cli/cli_udp.cpp [hunk 2]
```diff
@@ -120,6 +188,61 @@ template <> otError UdpExample::Process<Cmd("open")>(Arg aArgs[])
...
+ * - `type`/`value` combinations:
+ *   - `-t`: The payload in the `value` parameter is treated as text. If no `type` value
+ *   is entered, the payload in the `value` parameter is also treated as text.
+ *   - `-s`: Auto-generated payload with the specified length is given  in the `value` parameter.
```
Comment (abtink) [inline src/cli/cli_udp.cpp:240]:
```suggestion
 *   - `-s`: Auto-generated payload with the specified length as given in the `value` parameter.
```

File: src/cli/cli_udp.cpp [hunk 3]
```diff
@@ -120,6 +188,61 @@ template <> otError UdpExample::Process<Cmd("open")>(Arg aArgs[])
...
+ *   - `-x`: Binary data in hexadecimal representation in given in the `value` parameter.
```
Comment (abtink) [inline src/cli/cli_udp.cpp:241]:
```suggestion
 *   - `-x`: Binary data in hexadecimal representation is given in the `value` parameter.
```

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 
Couple of smaller  suggestions below.

File: src/cli/cli_udp.cpp [hunk 4]
```diff
@@ -50,6 +50,35 @@ UdpExample::UdpExample(otInstance *aInstance, OutputImplementer &aOutputImplemen
...
+ * @code
+ * udp bind -u :: 1234
+ * Done
+ * @endcode
+ * @code
+ * udp bind -b :: 1234
+ * Done
+ * @endcode
+ * @cparam udp bind [@ca{netif}] @ca{ip} @ca{port}
+ * - netif: The binding network interface, which is determined as follows:
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:68]: Can you make each parameter on each bullet code font please

- `netif`:
- `ip`:
- `port`:

File: src/cli/cli_udp.cpp [hunk 5]
```diff
@@ -77,6 +106,27 @@ template <> otError UdpExample::Process<Cmd("bind")>(Arg aArgs[])
...
+ * @endcode
+ * @code
+ * udp connect 172.17.0.1 1234
+ * Connecting to synthesized IPv6 address: fdde:ad00:beef:2:0:0:ac11:1
+ * Done
+ * @endcode
+ * @cparam udp connect @ca{ip} @ca{port}
+ * The following parameters are required:
+ * - `ip`: IP address of the peer
+ * - `port`: UDP port number of the peer
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:123]: Can you add a period at the end of each first sentence here

IP address of the peer.
UPD port number of the peer.

File: src/cli/cli_udp.cpp [hunk 6]
```diff
@@ -201,10 +324,33 @@ template <> otError UdpExample::Process<Cmd("linksecurity")>(Arg aArgs[])
 {
     otError error = OT_ERROR_NONE;
 
+    /**
+     * @cli udp linksecurity(Status)
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:328]: `@cli udp linksecurity`

We can drop the status on this one

File: src/cli/cli_udp.cpp [hunk 7]
```diff
@@ -201,10 +324,33 @@ template <> otError UdpExample::Process<Cmd("linksecurity")>(Arg aArgs[])
...
+     * @endcode
+     * @par
+     * Indicates whether link security is enabled or disabled.
+     */
     if (aArgs[0].IsEmpty())
     {
         OutputEnabledDisabledStatus(mLinkSecurityEnabled);
     }
+    /**
+     * @cli udp linksecurity(Enable,Disable)
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:342]: `@cli udp linksecurity (enable,disable)`

File: src/cli/cli_udp.cpp [hunk 8]
```diff
@@ -120,6 +188,61 @@ template <> otError UdpExample::Process<Cmd("open")>(Arg aArgs[])
...
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -x 68656c6c6f
+ * Done
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -s 800
+ * Done
+ * @endcode
+ * @cparam udp send [@ca{ip} @ca{port}] [@ca{type}] @ca{value}
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:230]: The way you have the parameters... does it mean that IP and Port must be specified together, but are also optional?  I.e. you wouldn't specify just the IP?

Can you make that clear in the description, that if you specify an IP you have to specify the Port.

Also, where does a message get sent if you don't specify an IP/port ... to all nodes in the Thread network?

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli_udp.cpp (hunk 8 above)
Comment (abtink) [reply src/cli/cli_udp.cpp:230]: the message is sent using the socket. If `[ip port]` are not given the socket peer addresses are used (from `udp connect`).

File: src/cli/cli_udp.cpp [hunk 9]
```diff
@@ -201,10 +329,33 @@ template <> otError UdpExample::Process<Cmd("linksecurity")>(Arg aArgs[])
...
+     * @endcode
+     * @par
+     * Indicates whether link security is enabled or disabled.
+     */
     if (aArgs[0].IsEmpty())
     {
         OutputEnabledDisabledStatus(mLinkSecurityEnabled);
     }
+    /**
+     * @cli udp linksecurity(enable,disable)
```
Comment (Vyrastas) [inline src/cli/cli_udp.cpp:347]: Space before the parenthesis, please

`@cli udp linksecurity (enable,disable)`

**Review by Vyrastas — CHANGES_REQUESTED** [review]
A small nit and then we're good

**Review by Vyrastas — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**